mh/session.py

```diff
@@ -197,6 +199,16 @@ def join_city(self, city_id):
     def leave_city(self):
         DB.leave_city(self)
 
+    def try_transfer_city_leadership(self):
+        city = self.get_city()
```

Comment:
This method should perform the layer check and return `None` early if the `local_info[city_id]` is `None`.

Reply:
`session.get_city()` already does that check with an assertion. This function should only be able to be called from in the city, and when I programmed it I figured the `assert` within `get_city()` would be enough to cover that base. I should instead do an explicit `None` check within this function in addition to the check within `get_city()`?

Reply:
My point was to avoid the assert to be triggered because it will raise an exception that will close the connection. The advantage of this approach is that you don't need to check for the layer prior using this method. On top of that, it makes more sense for a function with `try` in the name not to fail.

Reply:
Gotcha. This will be in the next commit.